server/tests/rider.test.ts

```diff
@@ -0,0 +1,587 @@
+      await generateGetRiderUsageTest(riderToken));
+    it('should fail with 400 given no authorization header', async () => {
+      const res = await request(app)
+        .get('/api/riders/abc-10/usage')
+        .expect(400);
+      expect(res.body).have.property('err');
+    });
+  });
+  // TODO: CREATE CURRENT RIDE TESTS
+  // testing retrieval of rider's current ride
+  describe("Testing Get Rider's Current/Soonest Rider Within 30 Minutes", () => {
+    const generateGetCurrRideTest = async (authToken: string) => {
+      const res = await request(app)
+        .get('/api/riders/abc-10/currentride')
+        .auth(authToken, { type: 'bearer' })
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      res.status === 200
+        ? (expect(res.status).to.be.equal(200),
+          expect(res.body).to.deep.equal(testRides[0]))
+        : expect(res.status).to.be.equal(400);
+    };
+    it('should return correct response for Admin account', async () =>
+      await generateGetCurrRideTest(adminToken));
+    it('should fail with 400 given Driver account', async () =>
+      await generateGetCurrRideTest(driverToken));
+    it('should return correct response for Rider account', async () =>
+      await generateGetCurrRideTest(riderToken));
+    it('should fail with 400 given no authorization header', async () => {
+      const res = await request(app)
+        .get('/api/riders/abc-10/currentride')
+        .expect(400);
+      expect(res.body).have.property('err');
+    });
+  });
+
+  // testing retrieval of specific rider by ID
+  describe('Testing Get Rider by Id', () => {
+    const generateGetRiderTest = async (authToken: string) => {
+      const res = await request(app)
+        .get('/api/riders/abc-10')
+        .auth(authToken, { type: 'bearer' })
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.status).to.be.equal(200);
+      expect(res.body).to.have.property('data');
+      expect(res.body.data).to.deep.equal(testRiders[0]);
+    };
+    it('should return correct response for Admin account', async () =>
+      await generateGetRiderTest(adminToken));
+    it('should return correct response for Driver account', async () =>
+      await generateGetRiderTest(driverToken));
+    it('should return correct response for Rider account', async () =>
+      await generateGetRiderTest(riderToken));
+    it('should fail with 400 given no authorization header', async () => {
+      const res = await request(app).get('/api/riders/abc-10').expect(400);
+      expect(res.body).have.property('err');
+    });
+  });
+
+  // testing retrieval of all riders
+  describe('Testing fetching all riders', () => {
+    const generateGetAllRidersTest = async (authToken: string) => {
+      const res = await request(app)
+        .get('/api/riders/')
+        .auth(authToken, { type: 'bearer' })
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      res.status === 200
```

Comment:
is there a specific reason you are doing a conditional test based on the status? I think it might cause problems in the future. If the endpoint is broken and returns `400 error`, it will still pass the test which we don't want to happen